lib/web_ui/lib/src/engine/html/path/path_ref.dart

```diff
@@ -173,9 +182,47 @@ class PathRef {
   }
 
   /// Reconstructs Rect from path commands.
-  ui.Rect _getRect() {
-    return ui.Rect.fromLTRB(
-        atPoint(0).dx, atPoint(0).dy, atPoint(1).dx, atPoint(2).dy);
+  ui.Rect? _detectRect() {
+    assert(_fVerbs[0] == SPath.kMoveVerb);
+    final double x0 = atPoint(0).dx;
+    final double y0 = atPoint(0).dy;
+    final double x1 = atPoint(1).dx;
+    final double y1 = atPoint(1).dy;
+    if (_fVerbs[1] != SPath.kLineVerb || y1 != y0) {
```

Comment:
Do we only detect rects that begin with a horizontal border?

Reply:
yes. added comment.